server/src/yml-objects/AbstractYmlObjectHelper.ts

```diff
@@ -0,0 +1,58 @@
+import { CompletionItemKind, DocumentSymbol, SymbolKind } from 'vscode-languageserver';
+
+import { AbstractYmlObject } from '.';
+
+/**
+ * Create a `DocumentSymbol` object from an `AbstractYmlObject`.
+ *
+ * @param elem the element used as a base
+ *
+ * @return a new DocumentSymbol
+ */
+export function completionItemToDocumentSymbol(elem: AbstractYmlObject) {
+    let _label = elem.label;
+    if (_label.includes('::')) {
+        _label = _label.substring(_label.lastIndexOf('::'));
+    }
+    return DocumentSymbol.create(
+        _label,
+        elem.detail,
+        completionItemKindToSymbolKind(elem.kind),
+        elem.definitionLocation.range,
+        elem.definitionLocation.range,
+    );
+}
+
+/**
+ * Translate `CompletionItemKind` values into `SymbolKind`. Both don't use the same symbols
+ * and whenthey do the value is often if not always different.
+ *
+ * @param completionItemKind the completion item kind to translate
+ *
+ * @returns an equivalent SymbolKind for the provided CompletionItemKind or `Property` if there is no match.
+ */
+export function completionItemKindToSymbolKind(completionItemKind: CompletionItemKind) {
+    switch (completionItemKind) {
```

Comment:
In Java I’d say “make a private static final unmodifiable EnumMap”. Dunno if you can have something reasonably close to it.
Kinda like https://github.com/yseop/com.yseop.suite/blob/78de55142e87c716b2c5f97daa11676e2ffc6a24/com.yseop.manager/src/main/java/com/yseop/security/PermissionAuthority.java#L234, except that it isn’t an EnumMap here.

Reply:
We don't have EnumMap types, but a simple Map should be enough.